metrics/core.go

```diff
@@ -1,119 +1,198 @@
+	for _, me := range m.MetricEngines {
+		me.RecordPutBackendXml()
+	}
+}
+
+func (m Metrics) RecordPutBackendJson() {
+	for _, me := range m.MetricEngines {
+		me.RecordPutBackendJson()
+	}
+}
+
+func (m Metrics) RecordPutBackendInvalid() {
+	for _, me := range m.MetricEngines {
+		me.RecordPutBackendInvalid()
+	}
+}
+
+func (m Metrics) RecordPutBackendDefTTL() {
+	for _, me := range m.MetricEngines {
+		me.RecordPutBackendDefTTL()
+	}
+}
+
+func (m Metrics) RecordPutBackendDuration(duration *time.Time) {
+	for _, me := range m.MetricEngines {
+		me.RecordPutBackendDuration(duration)
+	}
+}
+
+func (m Metrics) RecordPutBackendError() {
+	for _, me := range m.MetricEngines {
+		me.RecordPutBackendError()
+	}
+}
+
+func (m Metrics) RecordPutBackendSize(sizeInBytes float64) {
+	for _, me := range m.MetricEngines {
+		me.RecordPutBackendSize(sizeInBytes)
+	}
+}
+
+func (m Metrics) RecordGetBackendDuration(duration *time.Time) {
+	for _, me := range m.MetricEngines {
+		me.RecordGetBackendDuration(duration)
+	}
+}
+
+func (m Metrics) RecordGetBackendTotal() {
+	for _, me := range m.MetricEngines {
+		me.RecordGetBackendTotal()
+	}
+}
+
+func (m Metrics) RecordGetBackendError() {
+	for _, me := range m.MetricEngines {
+		me.RecordGetBackendError()
+	}
+}
+
+func (m Metrics) IncreaseOpenConnections() {
+	for _, me := range m.MetricEngines {
+		me.IncreaseOpenConnections()
+	}
+}
+
+func (m Metrics) DecreaseOpenConnections() {
+	for _, me := range m.MetricEngines {
+		me.DecreaseOpenConnections()
+	}
+}
+
+func (m Metrics) RecordCloseConnectionErrors() {
+	for _, me := range m.MetricEngines {
+		me.RecordCloseConnectionErrors()
+	}
+}
+
+func (m Metrics) RecordAcceptConnectionErrors() {
+	for _, me := range m.MetricEngines {
+		me.RecordAcceptConnectionErrors()
+	}
+}
+
+func (m Metrics) RecordExtraTTLSeconds(value float64) {
+	for _, me := range m.MetricEngines {
+		me.RecordExtraTTLSeconds(value)
+	}
+}
+
+func (m Metrics) Export(cfg config.Configuration) {
+	for _, me := range m.MetricEngines {
+		me.Export(cfg.Metrics)
+	}
+}
+
+// CacheMetrics Interface
+type CacheMetrics interface {
+	Export(cfg config.Metrics)
+	RecordPutError()
+	RecordPutBadRequest()
+	RecordPutTotal()
+	RecordPutDuration(duration *time.Time)
+	RecordGetError()
+	RecordGetBadRequest()
+	RecordGetTotal()
+	RecordGetDuration(duration *time.Time)
+	RecordPutBackendXml()
+	RecordPutBackendJson()
+	RecordPutBackendInvalid()
+	RecordPutBackendDefTTL()
+	RecordPutBackendDuration(duration *time.Time)
+	RecordPutBackendError()
+	RecordPutBackendSize(sizeInBytes float64)
+	RecordGetBackendTotal()
+	RecordGetBackendDuration(duration *time.Time)
+	RecordGetBackendError()
+	IncreaseOpenConnections()
+	DecreaseOpenConnections()
+	RecordCloseConnectionErrors()
+	RecordAcceptConnectionErrors()
+	RecordExtraTTLSeconds(value float64)
+}
+
+func CreateMetrics(cfg config.Configuration) *Metrics {
+	engineList := make([]CacheMetrics, 0, 2)
+
+	if cfg.Metrics.Influx.Host != "" {
```

Comment:
If it's correct that we've validated the metric config by this point, then consider using the .Enabled flag for both influx and prometheus.

Reply:
done